File: src/main/java/org/animefoda/topawardsbackend/entities/event/EventEntity.java
package org.animefoda.topawardsbackend.entities.event;

import jakarta.persistence.*;
import lombok.*;
import org.animefoda.topawardsbackend.entities.BaseEntity;

import java.time.OffsetDateTime;

@Entity
@Table(name = "event")
@Data
public class EventEntity extends BaseEntity<EventDTO> {
    @Column
    private String name;

    @Column(name = "start_date")
    private OffsetDateTime startDate;

    @Column(name = "end_date")
    private OffsetDateTime endDate;

    @Override
    public EventDTO toDTO() {
        return new EventDTO(id, name, startDate, endDate);
    }
}

File: src/main/java/org/animefoda/topawardsbackend/entities/event/EventEntity.kt
package org.animefoda.topawardsbackend.entities.event

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import org.animefoda.topawardsbackend.entities.BaseEntity
import java.time.LocalDate
import java.time.OffsetDateTime

@Entity
@Table(name = "event")
class EventEntity: BaseEntity<EventEntity,EventDTO>() {
    @Column
    var name = ""

    @Column(name = "start_date")
    var startDate: LocalDate = LocalDate.now()

    @Column(name = "end_date")
    var endDate: LocalDate = LocalDate.now().plusDays(1)

    public override fun toDTO(): EventDTO {
        return EventDTO(this.id, name, startDate, endDate)
    }
}